packages/swingset-liveslots/src/collectionManager.js

```diff
@@ -914,24 +909,88 @@ export function makeCollectionManager(
     );
   }
 
-  function reanimateScalarMapStore(vobjID) {
+  function reanimateMapStore(vobjID) {
     return collectionToMapStore(reanimateCollection(vobjID));
   }
 
-  function reanimateScalarWeakMapStore(vobjID) {
+  function reanimateWeakMapStore(vobjID) {
     return collectionToWeakMapStore(reanimateCollection(vobjID));
   }
 
-  function reanimateScalarSetStore(vobjID) {
+  function reanimateSetStore(vobjID) {
     return collectionToSetStore(reanimateCollection(vobjID));
   }
 
-  function reanimateScalarWeakSetStore(vobjID) {
+  function reanimateWeakSetStore(vobjID) {
     return collectionToWeakSetStore(reanimateCollection(vobjID));
   }
 
   const testHooks = { obtainStoreKindID, storeSizeInternal, makeCollection };
 
+  /**
+   * @param {Pattern} conjunct
+   * @param {StoreOptions} options
+   * @returns {StoreOptions}
+   */
+  const narrowKeyShapeOption = (conjunct, options) => {
+    const { keyShape: baseKeyShape } = options;
+    // To prepare for pattern-based compression
+    // https://github.com/Agoric/agoric-sdk/pull/6432
+    // put the substantive pattern, if any, last in the `M.and` since
+    // an `M.and` pattern compresses only according to its last conjunct.
+    const keyShape =
+      baseKeyShape === undefined ? conjunct : M.and(conjunct, baseKeyShape);
+    return harden({ ...options, keyShape });
+  };
```

Comment:
```suggestion
  /**
   * @param {Pattern} baseKeyShape
   * @param {StoreOptions} options
   * @returns {StoreOptions}
   */
  const narrowKeyShapeOption = (baseKeyShape, options) => {
    const { keyShape: keyShapeRestriction } = options;
    // To prepare for pattern-based compression
    // https://github.com/Agoric/agoric-sdk/pull/6432
    // put the substantive pattern, if any, last in the `M.and` since
    // an `M.and` pattern compresses only according to its last conjunct.
    const keyShape =
      keyShapeRestriction === undefined ? baseKeyShape : M.and(baseKeyShape, keyShapeRestriction);
    return harden({ ...options, keyShape });
  };
```

We might also consider a rename to `refineKeyShapeOption`, which has the (admittedly quite minor) benefit of not using a verb that can be confused for an adjective.

Reply:
Done.

I decided to keep "narrow" rather than "refine". The adjective reading leads in the same direction: it produces a narrower KeyShapeOption than it was given. To a formalist, the direction of "refine" is as clear. But to my non-formalist ears, the directionality of "narrow" is clearer.